services/web/server/src/simcore_service_webserver/projects/projects_utils.py

```diff
@@ -286,3 +294,69 @@ def any_node_inputs_changed(
                     )
                     return True
     return False
+
+
+def get_frontend_node_outputs_changes(
+    new_node: dict[str, Any], old_node: dict[str, Any], filter_keys: set[ServiceKey]
+) -> FrontendOutputsChanges:
+    """if node is a specific type it checks if outputs changed"""
+    nodes_keys = {old_node.get("key"), new_node.get("key")}
+    if not (len(nodes_keys) == 1 and nodes_keys.pop() in filter_keys):
+        return FrontendOutputsChanges(changed=False, keys=set())
```

Comment:
Please double check. Are you sure there is a ``not``  in front of this condition?  `` len(nodes_keys) ==1`` means ``new_node[key] ==  old_node[key]`` that they are the same so there are no changes, right?
 I wonder how it pass tests ... or perhaps I am missing something.

Perhaps using named variable might help reading the logic condition better?